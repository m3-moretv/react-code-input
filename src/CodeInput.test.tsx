import * as React from 'react';
import { shallow, mount } from 'enzyme';
import CodeInput, { Types } from './index';

function mockFunctions() {
  const original = require.requireActual('./utils.ts');
  return {
    ...original,
    uuid: jest.fn(() => '1')
  };
}

jest.mock('./utils.ts', () => mockFunctions());

const onChange: jest.Mock = jest.fn();

test('should render code input', () => {
  const wrapper = shallow(
    <CodeInput
      type={Types.text}
      fields={6}
      onChange={onChange}
      value=""
      name="text"
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should fires onchange with correct values', () => {
  const VALUE = '123';
  const wrapper = mount(
    <CodeInput
      onChange={onChange}
      fields={3}
      name="text"
      value={VALUE}
      type={Types.text}
    />
  );
  wrapper
    .find('input')
    .at(0)
    .simulate('change');

  expect(onChange).toHaveBeenCalledWith(VALUE);
});

test('should correctly handles backspace', () => {
  const VALUE = '123';
  const wrapper = mount(
    <CodeInput
      onChange={onChange}
      fields={3}
      name="text"
      value={VALUE}
      type={Types.text}
    />
  );

  wrapper
    .find('input')
    .at(0)
    .simulate('keydown', { keyCode: 8 });

  expect(
    wrapper
      .find('input')
      .at(0)
      .props().value
  ).toBe('');

  expect(onChange).toHaveBeenCalledWith(VALUE.substr(1));
});
