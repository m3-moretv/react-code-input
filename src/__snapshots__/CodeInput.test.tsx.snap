// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render code input 1`] = `
<div>
  <input
    className=""
    data-id={0}
    id="1"
    key="1"
    name="text"
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
  <input
    className=""
    data-id={1}
    id="1"
    key="1"
    name="text"
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
  <input
    className=""
    data-id={2}
    id="1"
    key="1"
    name="text"
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
  <input
    className=""
    data-id={3}
    id="1"
    key="1"
    name="text"
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
  <input
    className=""
    data-id={4}
    id="1"
    key="1"
    name="text"
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
  <input
    className=""
    data-id={5}
    id="1"
    key="1"
    name="text"
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
</div>
`;
